cfme/tests/infrastructure/test_timelines.py

```diff
@@ -3,163 +3,110 @@
+pytest_generate_tests = testgen.generate([InfraProvider], scope='module')
 
 
 @pytest.fixture(scope="module")
-def delete_fx_provider_event(db, provider):
-    logger.debug("Deleting timeline events for provider name %s", provider.name)
-    ems = db['ext_management_systems']
-    ems_events_table_name = version.pick({version.LOWEST: 'ems_events', '5.5': 'event_streams'})
-    ems_events = db[ems_events_table_name]
-    with db.transaction:
-        providers = (
-            db.session.query(ems_events.id)
-            .join(ems, ems_events.ems_id == ems.id)
-            .filter(ems.name == provider.name)
-        )
-        db.session.query(ems_events).filter(ems_events.id.in_(providers.subquery())).delete(False)
-
-
-def pytest_generate_tests(metafunc):
-    argnames, argvalues, idlist = testgen.providers_by_class(
-        metafunc, [VMwareProvider, RHEVMProvider])
-    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
-
-
-@pytest.fixture(scope="module")
-def vm_name():
-    # We have to use "tt" here to avoid name truncating in the timelines view
-    return "test_tt_" + fauxfactory.gen_alphanumeric(length=4)
-
-
-@pytest.mark.uncollectif(lambda: version.current_version() >= '5.7')
-@pytest.fixture(scope="module")
-def test_vm(request, provider, vm_name, setup_provider_modscope):
-    """Fixture to provision appliance to the provider being tested if necessary"""
-    navigate_to(InfraProvider, 'All')
+def test_vm(request, provider):
+    vm_name = "test_tl_" + fauxfactory.gen_alphanumeric(length=4)
     vm = VM.factory(vm_name, provider)
 
     request.addfinalizer(vm.delete_from_provider)
 
-    if not provider.mgmt.does_vm_exist(vm_name):
-        vm.create_on_provider(find_in_cfme=True, allow_skip="default")
-        vm.refresh_relationships()
+    if not provider.mgmt.does_vm_exist(vm.name):
+        logger.info("deploying %s on provider %s", vm.name, provider.key)
+        vm.create_on_provider(allow_skip="default", find_in_cfme=True)
     return vm
 
 
 @pytest.fixture(scope="module")
-def gen_events(delete_fx_provider_event, provider, test_vm):
+def gen_events(test_vm):
     logger.debug('Starting, stopping VM')
-    mgmt = provider.mgmt
+    mgmt = test_vm.provider.mgmt
     mgmt.stop_vm(test_vm.name)
     mgmt.start_vm(test_vm.name)
 
 
-def count_events(vm, nav_step):
-    try:
-        nav_step()
-    except ToolbarOptionGreyedOrUnavailable:
-        return 0
-    except NoSuchElementException:
-        vm.rediscover()
-        return 0
-
-    events = []
-    for event in jstimelines.events():
-        data = event.block_info()
-        if vm.name in data.values():
-            events.append(event)
-            if len(events) > 0:
-                return len(events)
-    return 0
-
-
-@pytest.mark.uncollectif(lambda: version.current_version() >= '5.7')
-@pytest.mark.meta(blockers=[1264183, 1281746])
-def test_provider_event(provider, gen_events, test_vm):
+def count_events(target, vm):
```

Comment:
**OPTIONAL:** Any reason to/not to put this in a common area? As the fn seems identical for both cloud and infra?